src/core/ui.js

```diff
@@ -47,6 +47,9 @@ function ariaDecorate(elem, ariaMap) {
 
 const respecUI = hyperHTML`<div id='respec-ui' class='removeOnSave' hidden></div>`;
 const menu = hyperHTML`<ul id=respec-menu role=menu aria-labelledby='respec-pill' hidden></ul>`;
+window.addEventListener("load", () => {
```

Comment:
This doesn't need to wait until `load` as we already have a `menu` defined.

Reply:
Surely, the menu is defined , but the list items of menu are still not loaded. So, the focusEls will go empty and the function won't work well
We have to use something else if we don't want this

Reply:
Oh, you're right. We probably want to use `window.respecIsReady` in that case, which indicates every ReSpec module is loaded.

Reply:
I have made this change to trapFocus function but I don't understand why focusElvs is going empty
```async function trapFocus(element) {
  await document.respecIsReady;
  const focusableEls = element.querySelectorAll(
    "a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input:not([disabled]), select:not([disabled])"
  ); ```
Any comments @saschanaz

Reply:
First of all, we can't use top level await as we want it to be first supported by browsers.

Second, sorry, `document.respecIsReady` is undefined as `core/ui` somehow runs before `core/respec-ready`.

Let's keep depending on `load` for now.